{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Example notebook showing pipeline for schizophrenia analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eerdem/CELLECT-revised'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# quick/dirty method to allow importing\n",
    "import os\n",
    "os.chdir(\"../\")\n",
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import constants\n",
    "from scripts import convert_output_to_dataframe\n",
    "from scripts import gene_set_enrichment_analysis\n",
    "from scripts import calculate_beta_correlation\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Correct pvalue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bonferroni\n"
     ]
    }
   ],
   "source": [
    "# type of pval correction (can be modified in the constants.py file)\n",
    "print(constants.PVAL_CORRECTION)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gwas</th>\n",
       "      <th>specificity_id</th>\n",
       "      <th>annotation</th>\n",
       "      <th>beta</th>\n",
       "      <th>beta_se</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>method</th>\n",
       "      <th>n_methods</th>\n",
       "      <th>pvalue_bonferroni</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SCZ_PGC3_2020</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>GABAergic.LGN_Inh_CTXN3</td>\n",
       "      <td>-0.017441</td>\n",
       "      <td>0.122443</td>\n",
       "      <td>0.556633</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SCZ_PGC3_2020</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>GABAergic.LGN_Inh_LAMP5</td>\n",
       "      <td>-0.129795</td>\n",
       "      <td>0.098972</td>\n",
       "      <td>0.905128</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SCZ_PGC3_2020</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>GABAergic.LGN_Inh_NTRK1</td>\n",
       "      <td>0.052677</td>\n",
       "      <td>0.121780</td>\n",
       "      <td>0.332673</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SCZ_PGC3_2020</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>GABAergic.LGN_Inh_TRPC4</td>\n",
       "      <td>-0.308010</td>\n",
       "      <td>0.136322</td>\n",
       "      <td>0.988061</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SCZ_PGC3_2020</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>Glutamatergic.LGN_Exc_BTNL9</td>\n",
       "      <td>-0.126593</td>\n",
       "      <td>0.182346</td>\n",
       "      <td>0.756227</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1576</th>\n",
       "      <td>SCZ_PGC3_2020</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Tongue.keratinocyte</td>\n",
       "      <td>-0.178351</td>\n",
       "      <td>0.069668</td>\n",
       "      <td>0.994761</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1577</th>\n",
       "      <td>SCZ_PGC3_2020</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Trachea.blood_cell</td>\n",
       "      <td>-0.288160</td>\n",
       "      <td>0.090981</td>\n",
       "      <td>0.999229</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1578</th>\n",
       "      <td>SCZ_PGC3_2020</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Trachea.endothelial_cell</td>\n",
       "      <td>-0.117255</td>\n",
       "      <td>0.085372</td>\n",
       "      <td>0.915182</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1579</th>\n",
       "      <td>SCZ_PGC3_2020</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Trachea.epithelial_cell</td>\n",
       "      <td>-0.346570</td>\n",
       "      <td>0.080956</td>\n",
       "      <td>0.999991</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1580</th>\n",
       "      <td>SCZ_PGC3_2020</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Trachea.mesenchymal_cell</td>\n",
       "      <td>-0.262969</td>\n",
       "      <td>0.066674</td>\n",
       "      <td>0.999960</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1581 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gwas   specificity_id                   annotation      beta  \\\n",
       "0     SCZ_PGC3_2020  Allen_human_LGN      GABAergic.LGN_Inh_CTXN3 -0.017441   \n",
       "1     SCZ_PGC3_2020  Allen_human_LGN      GABAergic.LGN_Inh_LAMP5 -0.129795   \n",
       "2     SCZ_PGC3_2020  Allen_human_LGN      GABAergic.LGN_Inh_NTRK1  0.052677   \n",
       "3     SCZ_PGC3_2020  Allen_human_LGN      GABAergic.LGN_Inh_TRPC4 -0.308010   \n",
       "4     SCZ_PGC3_2020  Allen_human_LGN  Glutamatergic.LGN_Exc_BTNL9 -0.126593   \n",
       "...             ...              ...                          ...       ...   \n",
       "1576  SCZ_PGC3_2020     tabula_muris          Tongue.keratinocyte -0.178351   \n",
       "1577  SCZ_PGC3_2020     tabula_muris           Trachea.blood_cell -0.288160   \n",
       "1578  SCZ_PGC3_2020     tabula_muris     Trachea.endothelial_cell -0.117255   \n",
       "1579  SCZ_PGC3_2020     tabula_muris      Trachea.epithelial_cell -0.346570   \n",
       "1580  SCZ_PGC3_2020     tabula_muris     Trachea.mesenchymal_cell -0.262969   \n",
       "\n",
       "       beta_se    pvalue   method  n_methods  pvalue_bonferroni  \n",
       "0     0.122443  0.556633  H-MAGMA          3                1.0  \n",
       "1     0.098972  0.905128  H-MAGMA          3                1.0  \n",
       "2     0.121780  0.332673  H-MAGMA          3                1.0  \n",
       "3     0.136322  0.988061  H-MAGMA          3                1.0  \n",
       "4     0.182346  0.756227  H-MAGMA          3                1.0  \n",
       "...        ...       ...      ...        ...                ...  \n",
       "1576  0.069668  0.994761    MAGMA          3                1.0  \n",
       "1577  0.090981  0.999229    MAGMA          3                1.0  \n",
       "1578  0.085372  0.915182    MAGMA          3                1.0  \n",
       "1579  0.080956  0.999991    MAGMA          3                1.0  \n",
       "1580  0.066674  0.999960    MAGMA          3                1.0  \n",
       "\n",
       "[1581 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# After running CELLECT with the GWAS and scRNA files the output\n",
    "# will be stored in out/CELLECT-{GWAS_name} folder\n",
    "cellect_out_dir = \"out/CELLECT-SCZ_PGC3_2020\"\n",
    "\n",
    "df_scz = convert_output_to_dataframe.make_df(cellect_out_dir)\n",
    "df_scz = convert_output_to_dataframe.pvalue_correction(df_scz, method=constants.PVAL_CORRECTION)\n",
    "display(df_scz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene-Set Enrichment Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ARCHS4_Tissues\n",
      "Aging_Perturbations_from_GEO_down\n",
      "Aging_Perturbations_from_GEO_up\n",
      "Allen_Brain_Atlas_down\n",
      "Allen_Brain_Atlas_up\n",
      "ClinVar_2019\n",
      "Disease_Signatures_from_GEO_down_2014\n",
      "Disease_Signatures_from_GEO_up_2014\n",
      "ENCODE_Histone_Modifications_2013\n",
      "ENCODE_Histone_Modifications_2015\n",
      "Epigenomics_Roadmap_HM_ChIP-seq\n",
      "GO_Biological_Process_2018\n",
      "GO_Cellular_Component_2018\n",
      "GO_Molecular_Function_2018\n",
      "GTEx_Tissue_Sample_Gene_Expression_Profiles_down\n",
      "GTEx_Tissue_Sample_Gene_Expression_Profiles_up\n",
      "GWAS_Catalog_2019\n",
      "HMDB_Metabolites\n",
      "Human_Gene_Atlas\n",
      "Jensen_COMPARTMENTS\n",
      "Jensen_DISEASES\n",
      "Jensen_TISSUES\n",
      "KEGG_2019_Human\n",
      "Reactome_2016\n",
      "Tissue_Protein_Expression_from_ProteomicsDB\n",
      "WikiPathways_2019_Human\n",
      "miRTarBase_2017\n"
     ]
    }
   ],
   "source": [
    "# gene-sets used in the analysis (can be modified in the constants.py file)\n",
    "for gene_set in constants.GENE_SET_LIST:\n",
    "    print(gene_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Performing GSEA...\n",
      "\n",
      "Top 5 ranked cell-types (N=68) in schizophrenia GWAS:\n",
      "1. Allen_human_MTG: Glutamatergic.Exc_L2_LAMP5_LTK (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU3 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH1 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU9 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU8 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU7 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU5 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU4 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU24 (N=1, freq=1.0)\n",
      "1. DroNc_Human_Hippocampus: GABA2 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU23 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU22 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU21 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU20 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU2 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU19 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH11 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH12 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH13 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH14 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH15 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH16 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH17 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH18 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH19 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH20 (N=1, freq=1.0)\n",
      "1. tabula_muris: Brain_Non-Myeloid.neuron (N=1, freq=1.0)\n",
      "1. tabula_muris: Brain_Non-Myeloid.oligodendrocyte (N=1, freq=1.0)\n",
      "1. tabula_muris: Brain_Non-Myeloid.oligodendrocyte_precursor_cell (N=1, freq=1.0)\n",
      "1. tabula_muris: Marrow.Slamf1-positive_multipotent_progenitor_cell (N=1, freq=1.0)\n",
      "1. tabula_muris: Marrow.common_lymphoid_progenitor (N=1, freq=1.0)\n",
      "1. tabula_muris: Marrow.granulocyte_monocyte_progenitor_cell (N=1, freq=1.0)\n",
      "1. tabula_muris: Pancreas.pancreatic_A_cell (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU18 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU17 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU15 (N=1, freq=1.0)\n",
      "1. mousebrain: HBGLU6 (N=1, freq=1.0)\n",
      "1. DroNc_Human_Hippocampus: exCA1 (N=1, freq=1.0)\n",
      "1. DroNc_Human_Hippocampus: exDG (N=1, freq=1.0)\n",
      "1. DroNc_Human_Hippocampus: exPFC1 (N=1, freq=1.0)\n",
      "1. DroNc_Human_Hippocampus: exPFC2 (N=1, freq=1.0)\n",
      "1. PsychENCODE_DER-22: Ex1 (N=1, freq=1.0)\n",
      "1. PsychENCODE_DER-22: Ex3e (N=1, freq=1.0)\n",
      "1. PsychENCODE_DER-22: Ex9 (N=1, freq=1.0)\n",
      "1. mousebrain: DEGLU1 (N=1, freq=1.0)\n",
      "1. mousebrain: DEGLU2 (N=1, freq=1.0)\n",
      "1. mousebrain: DEGLU3 (N=1, freq=1.0)\n",
      "1. mousebrain: DEGLU4 (N=1, freq=1.0)\n",
      "1. mousebrain: DEINH1 (N=1, freq=1.0)\n",
      "1. mousebrain: DEINH3 (N=1, freq=1.0)\n",
      "1. mousebrain: HBGLU5 (N=1, freq=1.0)\n",
      "1. mousebrain: HBGLU7 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU14 (N=1, freq=1.0)\n",
      "1. mousebrain: HBGLU9 (N=1, freq=1.0)\n",
      "1. mousebrain: HBINH5 (N=1, freq=1.0)\n",
      "1. mousebrain: MEGLU6 (N=1, freq=1.0)\n",
      "1. mousebrain: MEINH10 (N=1, freq=1.0)\n",
      "1. mousebrain: MEINH2 (N=1, freq=1.0)\n",
      "1. mousebrain: MSN1 (N=1, freq=1.0)\n",
      "1. mousebrain: MSN2 (N=1, freq=1.0)\n",
      "1. mousebrain: MSN3 (N=1, freq=1.0)\n",
      "1. mousebrain: MSN5 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU1 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU10 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU11 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU12 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU13 (N=1, freq=1.0)\n",
      "1. tabula_muris: Pancreas.type_B_pancreatic_cell (N=1, freq=1.0)\n",
      "\n",
      "\n",
      "Converting Ensembl ID to Gene Symbol...\n",
      "\n",
      "(1/68) Converting Allen_human_MTG, Glutamatergic.Exc_L2_LAMP5_LTK (181 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(2/68) Converting mousebrain, TEGLU3 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(3/68) Converting mousebrain, TEINH1 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(4/68) Converting mousebrain, TEGLU9 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(5/68) Converting mousebrain, TEGLU8 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(6/68) Converting mousebrain, TEGLU7 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(7/68) Converting mousebrain, TEGLU5 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(8/68) Converting mousebrain, TEGLU4 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(9/68) Converting mousebrain, TEGLU24 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(10/68) Converting DroNc_Human_Hippocampus, GABA2 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(11/68) Converting mousebrain, TEGLU23 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(12/68) Converting mousebrain, TEGLU22 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(13/68) Converting mousebrain, TEGLU21 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(14/68) Converting mousebrain, TEGLU20 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(15/68) Converting mousebrain, TEGLU2 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(16/68) Converting mousebrain, TEGLU19 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(17/68) Converting mousebrain, TEINH11 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(18/68) Converting mousebrain, TEINH12 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(19/68) Converting mousebrain, TEINH13 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(20/68) Converting mousebrain, TEINH14 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(21/68) Converting mousebrain, TEINH15 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(22/68) Converting mousebrain, TEINH16 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(23/68) Converting mousebrain, TEINH17 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(24/68) Converting mousebrain, TEINH18 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(25/68) Converting mousebrain, TEINH19 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(26/68) Converting mousebrain, TEINH20 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(27/68) Converting tabula_muris, Brain_Non-Myeloid.neuron (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(28/68) Converting tabula_muris, Brain_Non-Myeloid.oligodendrocyte (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(29/68) Converting tabula_muris, Brain_Non-Myeloid.oligodendrocyte_precursor_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(30/68) Converting tabula_muris, Marrow.Slamf1-positive_multipotent_progenitor_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(31/68) Converting tabula_muris, Marrow.common_lymphoid_progenitor (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(32/68) Converting tabula_muris, Marrow.granulocyte_monocyte_progenitor_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(33/68) Converting tabula_muris, Pancreas.pancreatic_A_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(34/68) Converting mousebrain, TEGLU18 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(35/68) Converting mousebrain, TEGLU17 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(36/68) Converting mousebrain, TEGLU15 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(37/68) Converting mousebrain, HBGLU6 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(38/68) Converting DroNc_Human_Hippocampus, exCA1 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(39/68) Converting DroNc_Human_Hippocampus, exDG (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(40/68) Converting DroNc_Human_Hippocampus, exPFC1 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(41/68) Converting DroNc_Human_Hippocampus, exPFC2 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(42/68) Converting PsychENCODE_DER-22, Ex1 (146 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(43/68) Converting PsychENCODE_DER-22, Ex3e (146 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(44/68) Converting PsychENCODE_DER-22, Ex9 (146 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(45/68) Converting mousebrain, DEGLU1 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(46/68) Converting mousebrain, DEGLU2 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(47/68) Converting mousebrain, DEGLU3 (151 genes)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(48/68) Converting mousebrain, DEGLU4 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(49/68) Converting mousebrain, DEINH1 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(50/68) Converting mousebrain, DEINH3 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(51/68) Converting mousebrain, HBGLU5 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(52/68) Converting mousebrain, HBGLU7 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(53/68) Converting mousebrain, TEGLU14 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(54/68) Converting mousebrain, HBGLU9 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(55/68) Converting mousebrain, HBINH5 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(56/68) Converting mousebrain, MEGLU6 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(57/68) Converting mousebrain, MEINH10 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(58/68) Converting mousebrain, MEINH2 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(59/68) Converting mousebrain, MSN1 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(60/68) Converting mousebrain, MSN2 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(61/68) Converting mousebrain, MSN3 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(62/68) Converting mousebrain, MSN5 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(63/68) Converting mousebrain, TEGLU1 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(64/68) Converting mousebrain, TEGLU10 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(65/68) Converting mousebrain, TEGLU11 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(66/68) Converting mousebrain, TEGLU12 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(67/68) Converting mousebrain, TEGLU13 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(68/68) Converting tabula_muris, Pancreas.type_B_pancreatic_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "Running Enrichr...\n",
      "\n",
      "(1/68) Analyzing Allen_human_MTG, Glutamatergic.Exc_L2_LAMP5_LTK (181 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(2/68) Analyzing mousebrain, TEGLU3 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(3/68) Analyzing mousebrain, TEINH1 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(4/68) Analyzing mousebrain, TEGLU9 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(5/68) Analyzing mousebrain, TEGLU8 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(6/68) Analyzing mousebrain, TEGLU7 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(7/68) Analyzing mousebrain, TEGLU5 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(8/68) Analyzing mousebrain, TEGLU4 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(9/68) Analyzing mousebrain, TEGLU24 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(10/68) Analyzing DroNc_Human_Hippocampus, GABA2 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(11/68) Analyzing mousebrain, TEGLU23 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(12/68) Analyzing mousebrain, TEGLU22 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(13/68) Analyzing mousebrain, TEGLU21 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(14/68) Analyzing mousebrain, TEGLU20 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(15/68) Analyzing mousebrain, TEGLU2 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(16/68) Analyzing mousebrain, TEGLU19 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(17/68) Analyzing mousebrain, TEINH11 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(18/68) Analyzing mousebrain, TEINH12 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(19/68) Analyzing mousebrain, TEINH13 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(20/68) Analyzing mousebrain, TEINH14 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(21/68) Analyzing mousebrain, TEINH15 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(22/68) Analyzing mousebrain, TEINH16 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(23/68) Analyzing mousebrain, TEINH17 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(24/68) Analyzing mousebrain, TEINH18 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(25/68) Analyzing mousebrain, TEINH19 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(26/68) Analyzing mousebrain, TEINH20 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(27/68) Analyzing tabula_muris, Brain_Non-Myeloid.neuron (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(28/68) Analyzing tabula_muris, Brain_Non-Myeloid.oligodendrocyte (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(29/68) Analyzing tabula_muris, Brain_Non-Myeloid.oligodendrocyte_precursor_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(30/68) Analyzing tabula_muris, Marrow.Slamf1-positive_multipotent_progenitor_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(31/68) Analyzing tabula_muris, Marrow.common_lymphoid_progenitor (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(32/68) Analyzing tabula_muris, Marrow.granulocyte_monocyte_progenitor_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(33/68) Analyzing tabula_muris, Pancreas.pancreatic_A_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(34/68) Analyzing mousebrain, TEGLU18 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(35/68) Analyzing mousebrain, TEGLU17 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(36/68) Analyzing mousebrain, TEGLU15 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(37/68) Analyzing mousebrain, HBGLU6 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(38/68) Analyzing DroNc_Human_Hippocampus, exCA1 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(39/68) Analyzing DroNc_Human_Hippocampus, exDG (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(40/68) Analyzing DroNc_Human_Hippocampus, exPFC1 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(41/68) Analyzing DroNc_Human_Hippocampus, exPFC2 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(42/68) Analyzing PsychENCODE_DER-22, Ex1 (146 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(43/68) Analyzing PsychENCODE_DER-22, Ex3e (146 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(44/68) Analyzing PsychENCODE_DER-22, Ex9 (146 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(45/68) Analyzing mousebrain, DEGLU1 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(46/68) Analyzing mousebrain, DEGLU2 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(47/68) Analyzing mousebrain, DEGLU3 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(48/68) Analyzing mousebrain, DEGLU4 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(49/68) Analyzing mousebrain, DEINH1 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(50/68) Analyzing mousebrain, DEINH3 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(51/68) Analyzing mousebrain, HBGLU5 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(52/68) Analyzing mousebrain, HBGLU7 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(53/68) Analyzing mousebrain, TEGLU14 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(54/68) Analyzing mousebrain, HBGLU9 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(55/68) Analyzing mousebrain, HBINH5 (151 genes)...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(56/68) Analyzing mousebrain, MEGLU6 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(57/68) Analyzing mousebrain, MEINH10 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(58/68) Analyzing mousebrain, MEINH2 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(59/68) Analyzing mousebrain, MSN1 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(60/68) Analyzing mousebrain, MSN2 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(61/68) Analyzing mousebrain, MSN3 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(62/68) Analyzing mousebrain, MSN5 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(63/68) Analyzing mousebrain, TEGLU1 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(64/68) Analyzing mousebrain, TEGLU10 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(65/68) Analyzing mousebrain, TEGLU11 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(66/68) Analyzing mousebrain, TEGLU12 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(67/68) Analyzing mousebrain, TEGLU13 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(68/68) Analyzing tabula_muris, Pancreas.type_B_pancreatic_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n"
     ]
    }
   ],
   "source": [
    "gwas_group_dict = {'schizophrenia':['SCZ']} # GWAS group name : [(regex) keywords]\n",
    "# the GWAS group name is just used for printing to stdout and can be anything\n",
    "# the keywords are used to find all gwas in the dataframe which contain the keywords\n",
    "gsea_dict = gene_set_enrichment_analysis.gsea(df_scz, gwas_group_dict)\n",
    "# since only one gwas (SCZ_PGC3_2020) is within the schizophrenia group all cell-types enriched have rank 1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cell-type Correlation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gwas</th>\n",
       "      <th>specificity_id</th>\n",
       "      <th>annotation</th>\n",
       "      <th>beta</th>\n",
       "      <th>beta_se</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>method</th>\n",
       "      <th>n_methods</th>\n",
       "      <th>pvalue_bonferroni</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>GABAergic.LGN_Inh_CTXN3</td>\n",
       "      <td>-0.174645</td>\n",
       "      <td>0.089013</td>\n",
       "      <td>0.975106</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>GABAergic.LGN_Inh_LAMP5</td>\n",
       "      <td>-0.123540</td>\n",
       "      <td>0.071732</td>\n",
       "      <td>0.957472</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>GABAergic.LGN_Inh_NTRK1</td>\n",
       "      <td>-0.269440</td>\n",
       "      <td>0.088436</td>\n",
       "      <td>0.998840</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>GABAergic.LGN_Inh_TRPC4</td>\n",
       "      <td>-0.246886</td>\n",
       "      <td>0.099982</td>\n",
       "      <td>0.993223</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>Glutamatergic.LGN_Exc_BTNL9</td>\n",
       "      <td>-0.421298</td>\n",
       "      <td>0.133637</td>\n",
       "      <td>0.999188</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879625</th>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Tongue.keratinocyte</td>\n",
       "      <td>-0.035636</td>\n",
       "      <td>0.045386</td>\n",
       "      <td>0.783817</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879626</th>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Trachea.blood_cell</td>\n",
       "      <td>-0.102483</td>\n",
       "      <td>0.059198</td>\n",
       "      <td>0.958279</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879627</th>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Trachea.endothelial_cell</td>\n",
       "      <td>-0.114722</td>\n",
       "      <td>0.055573</td>\n",
       "      <td>0.980498</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879628</th>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Trachea.epithelial_cell</td>\n",
       "      <td>-0.102770</td>\n",
       "      <td>0.052743</td>\n",
       "      <td>0.974312</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879629</th>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Trachea.mesenchymal_cell</td>\n",
       "      <td>0.050800</td>\n",
       "      <td>0.043447</td>\n",
       "      <td>0.121164</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>879630 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    gwas   specificity_id                   annotation  \\\n",
       "0         AD_JANSENS2019  Allen_human_LGN      GABAergic.LGN_Inh_CTXN3   \n",
       "1         AD_JANSENS2019  Allen_human_LGN      GABAergic.LGN_Inh_LAMP5   \n",
       "2         AD_JANSENS2019  Allen_human_LGN      GABAergic.LGN_Inh_NTRK1   \n",
       "3         AD_JANSENS2019  Allen_human_LGN      GABAergic.LGN_Inh_TRPC4   \n",
       "4         AD_JANSENS2019  Allen_human_LGN  Glutamatergic.LGN_Exc_BTNL9   \n",
       "...                  ...              ...                          ...   \n",
       "879625  volume: thalamus     tabula_muris          Tongue.keratinocyte   \n",
       "879626  volume: thalamus     tabula_muris           Trachea.blood_cell   \n",
       "879627  volume: thalamus     tabula_muris     Trachea.endothelial_cell   \n",
       "879628  volume: thalamus     tabula_muris      Trachea.epithelial_cell   \n",
       "879629  volume: thalamus     tabula_muris     Trachea.mesenchymal_cell   \n",
       "\n",
       "            beta   beta_se    pvalue   method  n_methods  pvalue_bonferroni  \n",
       "0      -0.174645  0.089013  0.975106  H-MAGMA          3                1.0  \n",
       "1      -0.123540  0.071732  0.957472  H-MAGMA          3                1.0  \n",
       "2      -0.269440  0.088436  0.998840  H-MAGMA          3                1.0  \n",
       "3      -0.246886  0.099982  0.993223  H-MAGMA          3                1.0  \n",
       "4      -0.421298  0.133637  0.999188  H-MAGMA          3                1.0  \n",
       "...          ...       ...       ...      ...        ...                ...  \n",
       "879625 -0.035636  0.045386  0.783817    MAGMA          3                1.0  \n",
       "879626 -0.102483  0.059198  0.958279    MAGMA          3                1.0  \n",
       "879627 -0.114722  0.055573  0.980498    MAGMA          3                1.0  \n",
       "879628 -0.102770  0.052743  0.974312    MAGMA          3                1.0  \n",
       "879629  0.050800  0.043447  0.121164    MAGMA          3                1.0  \n",
       "\n",
       "[879630 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of unqiue GWAS: 538\n"
     ]
    }
   ],
   "source": [
    "# at least two different gwas should be analysed prior to calculating the correlation\n",
    "# for this example a dataframe will be loaded including 500+ unqiue gwas\n",
    "\n",
    "df_all = pd.read_hdf('data/CELLECT_output/data.h5', 'df_all')\n",
    "\n",
    "display(df_all)\n",
    "print(f\"Number of unqiue GWAS: {df_all['gwas'].nunique()}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_df = calculate_beta_correlation.calculate_celltype_corr(df_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gwasx</th>\n",
       "      <th>gwasy</th>\n",
       "      <th>corr</th>\n",
       "      <th>pval</th>\n",
       "      <th>method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>ASD_2019</td>\n",
       "      <td>-0.220065</td>\n",
       "      <td>2.278355e-07</td>\n",
       "      <td>H-MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>BIP_PGC3</td>\n",
       "      <td>-0.122407</td>\n",
       "      <td>4.318468e-03</td>\n",
       "      <td>H-MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>BMI_GIANT2018</td>\n",
       "      <td>-0.007513</td>\n",
       "      <td>8.614700e-01</td>\n",
       "      <td>H-MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>DTI AD: ACR</td>\n",
       "      <td>0.247713</td>\n",
       "      <td>5.065423e-09</td>\n",
       "      <td>H-MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>DTI AD: ALIC</td>\n",
       "      <td>0.234532</td>\n",
       "      <td>3.299649e-08</td>\n",
       "      <td>H-MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144448</th>\n",
       "      <td>volume: caudate</td>\n",
       "      <td>volume: putamen</td>\n",
       "      <td>0.337825</td>\n",
       "      <td>6.184427e-16</td>\n",
       "      <td>MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144449</th>\n",
       "      <td>volume: caudate</td>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>0.351890</td>\n",
       "      <td>3.040520e-17</td>\n",
       "      <td>MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144450</th>\n",
       "      <td>volume: pallidum</td>\n",
       "      <td>volume: putamen</td>\n",
       "      <td>0.256756</td>\n",
       "      <td>1.313908e-09</td>\n",
       "      <td>MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144451</th>\n",
       "      <td>volume: pallidum</td>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>0.310798</td>\n",
       "      <td>1.330849e-13</td>\n",
       "      <td>MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144452</th>\n",
       "      <td>volume: putamen</td>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>0.229988</td>\n",
       "      <td>6.138297e-08</td>\n",
       "      <td>MAGMA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433359 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   gwasx             gwasy      corr          pval   method\n",
       "0         AD_JANSENS2019          ASD_2019 -0.220065  2.278355e-07  H-MAGMA\n",
       "1         AD_JANSENS2019          BIP_PGC3 -0.122407  4.318468e-03  H-MAGMA\n",
       "2         AD_JANSENS2019     BMI_GIANT2018 -0.007513  8.614700e-01  H-MAGMA\n",
       "3         AD_JANSENS2019       DTI AD: ACR  0.247713  5.065423e-09  H-MAGMA\n",
       "4         AD_JANSENS2019      DTI AD: ALIC  0.234532  3.299649e-08  H-MAGMA\n",
       "...                  ...               ...       ...           ...      ...\n",
       "144448   volume: caudate   volume: putamen  0.337825  6.184427e-16    MAGMA\n",
       "144449   volume: caudate  volume: thalamus  0.351890  3.040520e-17    MAGMA\n",
       "144450  volume: pallidum   volume: putamen  0.256756  1.313908e-09    MAGMA\n",
       "144451  volume: pallidum  volume: thalamus  0.310798  1.330849e-13    MAGMA\n",
       "144452   volume: putamen  volume: thalamus  0.229988  6.138297e-08    MAGMA\n",
       "\n",
       "[433359 rows x 5 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(corr_df)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
